src/state_machine.zig

```diff
@@ -103,28 +130,86 @@ pub fn StateMachineType(comptime Storage: type, comptime constants_: struct {
         const AccountsImmutableGroove = GrooveType(
             Storage,
             AccountImmutable,
+            .{
+                .timestamp = constants_.lsm_batch_multiple * math.max(
+                    constants.batch_max.create_accounts,
+                    // ×2 because creating a transfer will update 2 accounts.
+                    2 * constants.batch_max.create_transfers,
+                ),
+                .id = constants_.lsm_batch_multiple * constants.batch_max.create_accounts,
+                .user_data = constants_.lsm_batch_multiple * constants.batch_max.create_accounts,
+                .ledger = constants_.lsm_batch_multiple * constants.batch_max.create_accounts,
+                .code = constants_.lsm_batch_multiple * constants.batch_max.create_accounts,
```

Comment:
I don't feel strongly about it, but it doesn't feel like lsm_batch_multiple is actually a concern of the lsm, despite the naming. Especially once we have serial compaction - the contract for the lsm will just be that you must call compact every value_count_max operations and it won't actually know that batches exist.

Reply:
:+1: I'm fine leaving it out.

Aside, though: it is a little stricter than just "compact every value_count_max operations", since we have to make sure that data makes it to disk before the WAL overwrites it. (e.g. in the extreme case where each op only has one "put", we still need to compact since otherwise we might overwrite an entry in the WAL while some of those puts only exist in the mutable table).

Reply:
Agreed, I'm just saying the tree code doesn't know about that. The details batches and WAL are all external to the lsm. Which is kind of nice - less entangled.